pointcloud/pointcloud_utils.go

```diff
@@ -91,3 +91,22 @@ func StatisticalOutlierFilter(meanK int, stdDevThresh float64) (func(PointCloud)
 	}
 	return filterFunc, nil
 }
+
+// PointCloudToBasicOctree takes a pointcloud object and converts it into an octree.
+func PointCloudToBasicOctree(cloud PointCloud) (*BasicOctree, error) {
```

Comment:
If `cloud` is *already* a BasicOctree, we should return that directly instead of copying

Reply:
seems that is is not.
```
return utils.AssertType[*BasicOctree](cloud)
```
gives error
```
expected *pointcloud.BasicOctree but got *pointcloud.basicPointCloud
```

Unless, I am completely mis-understanding this comment.

Reply:
octrees still satify the pointcloud interface iirc, so a user could call this function with something thats already an octree. 

if we realize that we have an octree already, then we should just return it

Reply:
That would be because you passed in a `basicPointCloud`.

You should not be returning AssertType directly, only if the err is nil.

Reply:
PointCloud is an interface type -- there are many structs that fulfill the PointCloud interface. You should do a type assertion to see if the `cloud` passed is already of the BasicOctree Type, and return it if it is.

This isn't just for your use case, but a check so that all future users can ensure they're not accidently passing in a pointcloud that already an OctTree type

Reply:
```
if octTree, ok := cloud.(*BasicOctree); ok {
   return octTree, nil
}
```

https://yourbasic.org/golang/type-assertion-switch/